pkg/e2e/infra/utils.go

```diff
@@ -176,3 +291,57 @@ func getWorkerNode(client runtimeclient.Client) (*corev1.Node, error) {
 	}
 	return &nodeList.Items[0], nil
 }
+
+// areNodesReady returns true if an array of nodes are all ready
+func areNodesReady(nodes []*corev1.Node) bool {
+	// All nodes needs to be ready
+	for key := range nodes {
+		if !e2e.IsNodeReady(nodes[key]) {
+			glog.Errorf("Node %q is not ready. Conditions are: %v", nodes[key].Name, nodes[key].Status.Conditions)
+			return false
+		}
+		glog.Infof("Node %q is ready. Conditions are: %v", nodes[key].Name, nodes[key].Status.Conditions)
+	}
+	return true
+}
+
+// scaleMachineSet scales a machineSet with a given name to the given number of replicas
+func scaleMachineSet(name string, replicas int) error {
+	scaleClient, err := getScaleClient()
+	if err != nil {
+		return fmt.Errorf("error calling getScaleClient %v", err)
+	}
+
+	scale, err := scaleClient.Scales(e2e.TestContext.MachineApiNamespace).Get(schema.GroupResource{Group: machineAPIGroup, Resource: "MachineSet"}, name)
+	if err != nil {
+		return fmt.Errorf("error calling scaleClient.Scales get: %v", err)
+	}
+
+	scaleUpdate := scale.DeepCopy()
+	scaleUpdate.Spec.Replicas = int32(replicas)
```

Comment:
Shouldn't this be assigning a pointer? If so, how does the test currently work?

Reply:
`scaleUpdate.Spec.Replicas` is int32 type, pointer is for machine type